src/msg/async/ProtocolV1.cc

```diff
@@ -365,11 +363,12 @@ void ProtocolV1::write_event() {
         ack_left -= left;
         left = ack_left;
         r = connection->_try_send(left);
-      } else if (is_queued()) {
+      } else if (connection->is_queued()) { // data lefted in last write_event
         r = connection->_try_send();
       }
     }
 
+    connection->logger->tinc(l_msgr_send_messages_queue_lat, queue_lat);
```

Comment:
I don't think doing a single `tinc` call with the sum of the latencies is equivalent to make several `tinc` calls with each individual latency value. See perf_counter implementation here: https://github.com/ceph/ceph/blob/master/src/common/perf_counters.cc#L243